### PR TITLE
Added [directory] argument to init.

### PR DESCRIPTION
**Pull request by daniel-levin**
- git init supports providing an optional directory in which to
initialize a new repo
- the semantics is otherwise the same

### CODE REVIEW

**Comment by Byron** [comment]
Thanks a lot!
Could you have a look at the journey test and add test it works correctly there? It can be run with `make journey-test`.

**Comment by daniel-levin** [comment]
I now see how the automated tests work. I had to run `make journey-tests`.

File: git-repository/src/init.rs [hunk 1]
```diff
@@ -91,11 +91,17 @@ fn write_file(data: &[u8], path: &Path) -> Result<(), Error> {
...
-    fs::create_dir(p).map_err(|e| Error::CreateDirectory(e, p.to_owned()))
+    fs::create_dir_all(p).map_err(|e| Error::CreateDirectory(e, p.to_owned()))
 }
 
-pub fn repository() -> Result<(), Error> {
-    let mut cursor = PathBuf::from(GIT_DIR_NAME);
+pub fn repository(directory: Option<PathBuf>) -> Result<(), Error> {
+    let mut cursor = match directory {
+        Some(dir) => PathBuf::from(dir),
+        None => PathBuf::from(""),
```
Comment (edward-shen) [inline git-repository/src/init.rs:100]: nit: use `PathBuf::new()` here instead.

**Review by edward-shen — APPROVED** [review]
LGTM, unless @Byron has anything else to add.

**Review by Byron — APPROVED** [review]
Thanks a bunch!

I hope you liked the experience thus far and continue to contribute :).
Should there be a next time, you could try adding [a journey test like this](https://github.com/Byron/gitoxide/pull/60/commits/5c2fe3ae3b965110200d16cf5b90f165557a52ca), it really streamlines development by taking the manual testing out of your hands (especially with `make continuous-journey-tests`).

**Pull request merged**